amethyst_window/src/bundle.rs

```diff
@@ -45,6 +53,43 @@ impl WindowBundle {
 
         WindowBundle::from_config(display_config)
     }
+
+    fn build_event_loop_and_window(self) -> (Receiver<Event<()>>, Window) {
+        let (sender, receiver) = mpsc::channel::<Event<()>>();
+
+        let window_return: Arc<Mutex<Option<Window>>> = Arc::new(Mutex::new(None));
+        let done = Arc::new(AtomicBool::from(false));
+
+        Self::spawn_event_loop(self.config, sender, window_return.clone(), done.clone());
+
+        while !done.load(Ordering::SeqCst) {}
```

Comment:
probably better to sleep 1 ms instead of having a busy loop?

Reply:
Some platforms require event loop on main thread, so this logic won't work.

Consider taking a look for #1877